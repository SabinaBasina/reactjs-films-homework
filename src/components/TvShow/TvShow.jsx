import React, { PureComponent } from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import styles from './TvShow.scss';
import NoImage from './NoImage.jpg';

class TvShow extends PureComponent {
  render() {
    const { data } = this.props;
    return (
      <Link to={`/tvShowDetails/${data.id}`} className={styles.TvShow}>
        <br />
        <img src={data.image ? data.image.medium : NoImage} alt="" />
        {data.rating && data.rating.average
        && (
        <button className={styles.RatingTvShow} type="button">
          {data.rating.average}
        </button>
        )}
        <p>
          {data.name}
          <br />
          <span>
            {data.genres && Array.prototype.join.call(data.genres, ', ')}
          </span>
        </p>
      </Link>
    );
  }
}


export default TvShow;

TvShow.propTypes = {
  data: PropTypes.string,
};

TvShow.defaultProps = {
  data: 'undefined',
};
