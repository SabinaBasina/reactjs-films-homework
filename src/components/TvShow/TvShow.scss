.TvShow {
  text-decoration: none;
}

.TvShow p {
  color: black;
  font-size: 16px;
}

.TvShow p span {
  color: rgb(124, 124, 124);
  font-size: 14px;
}

.TvShow .RatingTvShow {
  position: absolute;
  width: 46px;
  height: 46px;
  color: rgb(255, 255, 255);
  background-color: rgb(235, 145, 55);
  font-size: 16px;
  border-radius: 25px;
  border: none;
  margin-top: 226px;
  margin-left: -22px;
}
