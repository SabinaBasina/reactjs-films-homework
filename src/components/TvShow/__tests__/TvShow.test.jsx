import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import TvShow from '../TvShow';

/* global test, expect */

test('TvShow test img is', () => {
  const renderer = new ShallowRenderer();
  renderer.render(<TvShow data={{ image: { medium: 'medium' } }} />);
  const result = renderer.getRenderOutput();

  expect(result).toMatchSnapshot();
});

test('TvShow test img no', () => {
  const renderer = new ShallowRenderer();
  renderer.render(<TvShow data={{ }} />);
  const result = renderer.getRenderOutput();

  expect(result).toMatchSnapshot();
});

test('TvShow test genres', () => {
  const renderer = new ShallowRenderer();
  renderer.render(<TvShow data={{ genres: 'genres' }} />);
  const result = renderer.getRenderOutput();

  expect(result).toMatchSnapshot();
});

test('TvShow test rating', () => {
  const renderer = new ShallowRenderer();
  renderer.render(<TvShow data={{ rating: { average: 'rating' } }} />);
  const result = renderer.getRenderOutput();

  expect(result).toMatchSnapshot();
});
