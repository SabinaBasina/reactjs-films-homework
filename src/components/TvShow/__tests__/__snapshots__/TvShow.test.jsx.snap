// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TvShow test genres 1`] = `
<Link
  className="TvShow"
  replace={false}
  to="/tvShowDetails/undefined"
>
  <br />
  <img
    alt=""
    src="test-file-stub"
  />
  <p>
    <br />
    <span>
      g, e, n, r, e, s
    </span>
  </p>
</Link>
`;

exports[`TvShow test img is 1`] = `
<Link
  className="TvShow"
  replace={false}
  to="/tvShowDetails/undefined"
>
  <br />
  <img
    alt=""
    src="medium"
  />
  <p>
    <br />
    <span />
  </p>
</Link>
`;

exports[`TvShow test img no 1`] = `
<Link
  className="TvShow"
  replace={false}
  to="/tvShowDetails/undefined"
>
  <br />
  <img
    alt=""
    src="test-file-stub"
  />
  <p>
    <br />
    <span />
  </p>
</Link>
`;

exports[`TvShow test rating 1`] = `
<Link
  className="TvShow"
  replace={false}
  to="/tvShowDetails/undefined"
>
  <br />
  <img
    alt=""
    src="test-file-stub"
  />
  <button
    className="RatingTvShow"
    type="button"
  >
    rating
  </button>
  <p>
    <br />
    <span />
  </p>
</Link>
`;
